### PR TITLE
1601 second solution

### PR DESCRIPTION
**Pull request by mohamedmorsy**
This PR solves the model resources versioning problem with ManyToMany relation regarding issue #1601. there are other resource types use ManyToMany relation and require the same fix. @zhiyuli I think web app resources may require the same fix.

### CODE REVIEW

**Comment by mohamedmorsy** [comment]
@pkdash could you please review?

**Comment by pkdash** [comment]
@mohamedmorsy Looks like you have to fix some failing unit tests.

**Comment by mohamedmorsy** [comment]
@pkdash this should be ready now to review. Thanks.

File: hs_swat_modelinstance/models.py [hunk 1]
```diff
@@ -46,12 +46,18 @@ def get_swat_model_objectives(self):
 
     @classmethod
     def _add_swat_objective(cls, model_objective, objectives):
```
Comment (pkdash) [inline hs_swat_modelinstance/models.py]: Looking at the code it seems this method is being called both in ```create()``` and ```update()``` functions of the ```ModelObjective``` class. I am wondering how the validation is successful in the function ```_validate_swat_model_objectives()``` when the list of values passed into this function are of type integers.

**Comment by pkdash** [comment]
@mohamedmorsy Left a comment. Once you address that comment, this should be ready to merge.

File: hs_swat_modelinstance/models.py (hunk 1 above)
Comment (mohamedmorsy) [inline hs_swat_modelinstance/models.py]: @pkdash The validation was created for the users who are using the rest api to create the resource to make sure that they pass the correct string. The validation method validates the values from creating a new resourcre using the normal way or the rest api as both of them pass string values. After fixing the versioning problem, now these integer values pass but ignored in the validation method. Do we really need to validate  that? If yes, any suggestions to enhance this?

Comment (pkdash) [inline hs_swat_modelinstance/models.py]: @mohamedmorsy 

> After fixing the versioning problem, now these integer values pass but ignored in the validation method

I not sure what you mean by 'ignored in the validation method'. Can you explain how it is ignored?

Comment (mohamedmorsy) [inline hs_swat_modelinstance/models.py]: @pkdash I mean I validate only the values of basestring

Comment (pkdash) [inline hs_swat_modelinstance/models.py]: @mohamedmorsy OK now I see that you have implemented that check for the swat_model_instance. What about mudflow_instance?. The ```_validate_params()``` need to do similar check.

Comment (mohamedmorsy) [inline hs_swat_modelinstance/models.py]: @pkdash yes exactly, the modflow do the same check

Comment (pkdash) [inline hs_swat_modelinstance/models.py]: @mohamedmorsy Sorry somehow I did not see that before.

**Comment by pkdash** [comment]
+1

**Comment by mohamedmorsy** [comment]
@pkdash Thanks!

**Pull request merged**